common/common.go

```diff
@@ -280,6 +282,25 @@ func GetLightdInfo() (*walletrpc.LightdInfo, error) {
 		saplingHeight = saplingJSON.ActivationHeight
 	}
 
+	// Report the most imminent (next) network upgrade (its name and height).
+	// First, copy all upgrades into a slice for sorting.
+	upgradesList := make([]Upgradeinfo, 0, len(getblockchaininfoReply.Upgrades))
+	for _, upgrade := range getblockchaininfoReply.Upgrades {
+		upgradesList = append(upgradesList, upgrade)
+	}
+	// Sort by height, ascending order.
+	sort.Slice(upgradesList, func(i, j int) bool {
+		return upgradesList[i].ActivationHeight < upgradesList[j].ActivationHeight
+	})
+	// Find the nearest (lowest-height)_pending upgrade, if any.
+	upgrade := Upgradeinfo{"", 0, ""}
+	for _, u := range upgradesList {
+		if u.Status == "pending" {
+			upgrade = u
+			break
+		}
+	}
```

Comment:
```suggestion
	// Find the name and activation height of the next pending network upgrade,
	// or "" and 0 if there is no pending upgrade.
	upgrade := Upgradeinfo{"", 0, ""}
	for _, u := range getblockchaininfoReply.Upgrades {
	    if u.Status == "pending" && (upgrade.Status == "" || u.ActivationHeight < upgrade.ActivationHeight) {
	        upgrade = u
	    }
	}
```
As well as being simpler, this is $O(n)$ rather than $O(n \log n)$.